src/pmdas/llc/llc.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Llc, configurable PMDA
+ *
+ * Copyright (c) 2023 Red Hat.
+ * 
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+
+#include <pcp/pmapi.h>
+#include <pcp/pmda.h>
+#include "domain.h"
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+/*
+ * LLC  Last-Level Cache of CPU info.
+ *
+ *  Linux supports resctrl which can be used to
+ *  control QoS of CPU cache and monitor LLC(last level cache).
+ *  This feature depends on the 'resctrl' pseudo filesystem.
+ *  The kernel should be built with the relevant config and
+ *  the pseudo-filesystem needs to be mounted:
+ *  mount -t resctrl resctrl -o mba_MBps /sys/fs/resctrl (on Intel)
+ *  mount -t resctrl resctrl -o cdp      /sys/fs/resctrl (on AMD)
+ *
+ *  Which results in directories like:
+ *   /sys/fs/resctrl/mon_data/mon_L3_XX
+ *   
+ *  These counters can be found in doc:
+ *   https://github.com/torvalds/linux/blob/master/Documentation/x86/resctrl.rst
+ *
+ * Metrics
+ *   llc.occupancy % of LLC in use
+ *   llc.mbm_local local memory bandwidth of this LLC
+ *   llc.mbm_total total memory bandwidth of this LLC
+ *
+ */
+
+/*
+ * instance domains
+ */
+
+const int LLC_INDOM = 0;
+
+static pmdaIndom indomtab[] = {
+    { LLC_INDOM, 0, NULL }
```

Comment:
It looks like (from CI) some C compilers are struggling with this ... might be the 'const int', I think most PMDAs use a \#define for this reason